### PR TITLE
Extend PaSh's expansion to properly handle input arguments

### PR DESCRIPTION
**Pull request by angelhof**
@mgree whenever you have some time you could take a look and see if that looks reasonable to you, no rush!

The input arguments for scripts are stored in the pash runtime in the `pash_input_args` variable and therefore expansion needs to search them there.

### CODE REVIEW

File: evaluation/tests/interface_tests/run.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ export PASH_TOP=${PASH_TOP:-$(git rev-parse --show-toplevel --show-superproject-
 
 ## TODO: Make the compiler work too.
```
[the remark this answers is not in the document]
Comment (angelhof) [reply evaluation/tests/interface_tests/run.sh:5]: Not anymore, good catch, it refers to the change that happened below!

**Review by mgree — APPROVED** [review]
Left comments in; lgtm given pending fixes.

**Comment by angelhof** [comment]
@nvasilakis Are these tests expected to fail?

**Comment by dkarnikis** [comment]
@angelhof These tests are not expected to fail. As mentioned here #286 , the annotation changes (555c3eb40a76a7c1afa94c180e3a8beea20b8b31) broke these two tests.

**Comment by angelhof** [comment]
Then we can merge the PR I guess right (since these failures are independent) :)

**Pull request merged**